### PR TITLE
Nazanin/remove trading times request when not needed

### PR DESCRIPTION
**Pull request by nazaninreihani**

### CODE REVIEW

File: app/index.jsx [hunk 1]
```diff
@@ -280,6 +280,7 @@ class App extends Component {
                 granularity={this.state.granularity}
                 onSettingsChange={this.saveSettings}
                 isConnectionOpened={isConnectionOpened}
+                noTradingTimesForStartEpoch
```
Comment (morteza-binary) [inline app/index.jsx]: @nazaninreihani It'd be better to change the name to `shouldFetchTradingTimes` and default to `True`.
```suggestion
                shouldFetchTradingTimes
```

File: src/binaryapi/TradingTimes.js [hunk 2]
```diff
@@ -8,7 +8,8 @@ class TradingTimes {
     isInitialized = false;
     tradingTimesPromise = new PendingPromise();
 
-    constructor(api) {
+    constructor(api, needsNoTradingTimes) {
```
Comment (morteza-binary) [inline src/binaryapi/TradingTimes.js]:
```suggestion
    constructor(api, shouldFetchTradingTimes = true) {
```

File: src/binaryapi/TradingTimes.js [hunk 3]
```diff
@@ -8,7 +8,8 @@ class TradingTimes {
...
+        this._needsNoTradingTimes = needsNoTradingTimes;
```
Comment (morteza-binary) [inline src/binaryapi/TradingTimes.js]:
```suggestion
        this._shouldFetchTradingTimes = shouldFetchTradingTimes;
```

File: src/binaryapi/TradingTimes.js [hunk 4]
```diff
@@ -28,7 +29,7 @@ class TradingTimes {
 
         this.lastUpdateDate = this._serverTime.getLocalDate().toISOString().substring(0, 10);
 
-        if (!this._tradingTimesMap) {
+        if (!this._tradingTimesMap && !this._needsNoTradingTimes) {
```
Comment (morteza-binary) [inline src/binaryapi/TradingTimes.js]:
```suggestion
        if (!this._tradingTimesMap && this._shouldFetchTradingTimes) {
```

File: src/store/ChartState.js [hunk 5]
```diff
@@ -25,6 +25,7 @@ class ChartState {
     @observable isChartClosed = false;
     @observable shouldMinimiseLastDigits = false;
     @observable isStaticChart = false;
+    @observable noTradingTimesForStartEpoch = false;
```
Comment (morteza-binary) [inline src/store/ChartState.js]:
```suggestion
    @observable shouldFetchTradingTimes = true;
```

File: src/store/ChartStore.js [hunk 6]
```diff
@@ -372,9 +372,10 @@ class ChartStore {
             onSettingsChange,
         } = props;
         this.api = new BinaryAPI(requestAPI, requestSubscribe, requestForget);
-
+        const needsNoTradingTimes = this.mainStore.state.endEpoch
```
Comment (morteza-binary) [inline src/store/ChartStore.js]: @nazaninreihani IMO, the chart shouldn't decide to fetch the trading times or not, it just should rely on the passing conditions.

File: src/store/ChartTitleStore.js [hunk 7]
```diff
@@ -61,7 +61,7 @@ export default class ChartTitleStore {
     @computed get isShowChartPrice() { return this.mainStore.chart.isChartAvailable; }
     @computed get tradingTimes() { return this.mainStore.chart.tradingTimes; }
     @computed get symbolOpenTime() {
-        const times = this.tradingTimes._tradingTimesMap[this.currentSymbol.symbol].times;
+        const times = this.tradingTimes._tradingTimesMap ? this.tradingTimes._tradingTimesMap[this.currentSymbol.symbol].times : [];
```
Comment (morteza-binary) [inline src/store/ChartTitleStore.js]:
```suggestion
        const times = this.tradingTimes._tradingTimesMap && this.tradingTimes._tradingTimesMap.length ? this.tradingTimes._tradingTimesMap[this.currentSymbol.symbol].times : [];
```

File: app/index.jsx (hunk 1 above)
Comment (nazaninreihani) [reply app/index.jsx]: yesss thanks, I couldn't think of a name for it I was hoping you suggest a name :-D

File: src/store/ChartStore.js (hunk 6 above)
Comment (nazaninreihani) [reply src/store/ChartStore.js]: you mean we handle it in deriv? for example we make it false when we have startepoch/endepoch?

**Comment by ashkanx** [comment]
@nazaninreihani conflicts

**Comment by nazaninreihani** [comment]
> @nazaninreihani conflicts

resolved :)

**Pull request merged**